src/flux/period.coffee

```diff
@@ -0,0 +1,44 @@
+# coffeelint: disable=no_empty_functions
+{CrudConfig, makeSimpleStore, extendConfig} = require './helpers'
+{CourseActions} = require './course'
+_ = require 'underscore'
+
+PeriodConfig = {
+
+  create: (courseId, params) ->
+
+  created: (period, courseId) ->
+    CourseActions.load(courseId)
+
+  save: (courseId, period, params) ->
+
+  saved: (period, courseId) ->
+    CourseActions.load(courseId)
+
+  delete: (period, courseId) ->
+
+  deleted: (period, courseId) ->
+    CourseActions.load(courseId)
+    
+
+  exports:
+
+    validatePeriodName: (name, periods) ->
+      error = ''
+      maxLength = 20
```

Comment:
it is arbitrary, can change to what restrictions we want on period names, if any.